test/error/bad_host_alignment.cpp

```diff
@@ -0,0 +1,35 @@
+#include "Halide.h"
+#include <stdio.h>
+
+using namespace Halide;
+using namespace Halide::Internal;
+
+IRPrinter irp(std::cerr);
+int main(int argc, char **argv) {
+    Func f;
+    Var x, y;
+    uint8_t a[10][10];
+
+    buffer_t buf;
+    buf.host = &a[0][0];
+    buf.extent[0] = 10;
+    buf.extent[1] = 10;
+    buf.stride[0] = 1;
+    buf.stride[1] = 10;
+    buf.elem_size = 1;
+    buf.min[0] = 0;
+    buf.min[1] = 0;
+    Buffer param_buf(UInt(8), &buf);
+    ImageParam in(UInt(8), 2);
+
+    in.set_host_alignment(512);
```

Comment:
Sorry for the thrash, but I was thinking about this more. The problem is right now, if the alignment assert doesn't fail, then it will run the code, and probably crash, trying to dereference the address 1. So, I think we need to change this again to allocate a buffer aligned to some value, then shift the pointer by 1. It's probably good enough to declare an array of integers (sizeof(int) = 4) on the stack, and then increment a pointer to that array by 1 byte.